src/Tree.tsx

```diff
@@ -783,12 +787,41 @@ class Tree<TreeDataType extends BasicDataNode = DataNode> extends React.Componen
   };
 
   onNodeClick: NodeMouseEventHandler = (e, treeNode) => {
-    const { onClick } = this.props;
+    const { onClick, expandAction } = this.props;
+    const { expandedKeys, flattenNodes } = this.state;
+    const { expanded, key } = treeNode;
+
+    if (expandAction === 'click') {
+      const node = flattenNodes.filter(nodeItem => nodeItem.key === key)[0];
+      const eventNode = convertNodePropsToEventData({
+        ...getTreeNodeProps(key, this.getTreeNodeRequiredProps()),
+        data: node.data,
+        active: true,
+      });
+
+      this.setExpandedKeys(expanded ? arrDel(expandedKeys, key) : arrAdd(expandedKeys, key));
+      this.onNodeExpand(e as React.MouseEvent<HTMLDivElement>, eventNode);
+    }
+
     onClick?.(e, treeNode);
   };
 
   onNodeDoubleClick: NodeMouseEventHandler = (e, treeNode) => {
-    const { onDoubleClick } = this.props;
+    const { onDoubleClick, expandAction } = this.props;
+    const { expandedKeys, flattenNodes } = this.state;
+    const { expanded, key } = treeNode;
+
+    if (expandAction === 'doubleClick') {
```

Comment:
```suggestion
    if (expandAction === 'dblclick') {
```

https://developer.mozilla.org/en-US/docs/Web/API/Element/dblclick_event